proto/networkpolicy.proto

```diff
@@ -0,0 +1,107 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package istio.workload;
+option go_package="pkg/workloadapi";
+
+import "google/protobuf/empty.proto";
+
+message RBAC {
+  string name = 1;
+  string namespace = 2;
+
+  // Determine the scope of this RBAC policy.
+  // If set to NAMESPACE, the 'namespace' field value will be used.
+  RBACScope scope = 3;
+  // The action to take if the request is matched with the rules.
+  // Default is ALLOW if not specified.
+  RBACPolicyAction action = 4;
+  // Set of RBAC policy groups each containing its rules.
+  // If at least one of the groups is matched the policy action will
+  // take place.
+  // Groups are OR-ed.
+  repeated RBACPolicyRulesGroup groups = 5;
+}
+
+message RBACPolicyRulesGroup {
+  // Rules are AND-ed
+  // This is a generic form of the authz policy's to, from and when
+  repeated RBACPolicyRules rules = 1;
+}
+
+message RBACPolicyRules {
+  // The logical behavior between the matches (if there are more than one)
+  //  MatchBehavior match_behavior = 1;
+  repeated RBACPolicyRuleMatch matches = 2;
+}
+
+message RBACPolicyRuleMatch {
+  // Values of specific type are ORed
+  // If multiple types are set, they are ANDed
+
+  repeated StringMatch namespaces = 1;
+  repeated StringMatch not_namespaces = 2;
```

Comment:
No need using stringMatch for namespaces, makes things a little more complex

Reply:
It's part of AurhorizationPolicy unfortunately. I agree but we can't really change it at this point